sdk/provisioning/Azure.Provisioning/tests/Expressions/InterpolatedStringExpressionTest.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Azure.Provisioning.Expressions;
+using NUnit.Framework;
+
+namespace Azure.Provisioning.Tests.Expressions
+{
+    public class InterpolatedStringExpressionTest
+    {
+        [TestCaseSource(nameof(BicepFunctionInterpolateTestData))]
+        public string ValidateBicepFunctionInterpolate(BicepValue<string> expression)
+        {
+            return expression.ToString();
+        }
+
+        private static IEnumerable<TestCaseData> BicepFunctionInterpolateTestData()
+        {
+            // test provisionable variable
+            yield return new TestCaseData(
+                BicepFunction.Interpolate(
+                    $"Var={new ProvisioningVariable("foo", typeof(string))}"
+                    ))
+                .Returns("'Var=${foo}'");
+
+            // test index expression in interpolation
+            yield return new TestCaseData(
+                BicepFunction.Interpolate(
+                    $"Endpoint={new IndexExpression(
+                        new IdentifierExpression("foo"),
+                        new StringLiteralExpression("bar")
+                    )}"))
+                .Returns("'Endpoint=${foo['bar']}'");
+        }
```

Comment:
Style nit - I generally avoid TestCaseSource like this because it makes it harder to debug when something fails.  Sometimes it can be worth it if you have _a lot_ of test data, but in this case I don't think there's too much repetition with just:

```suggestion
    [Test]
    public void ValidateBicepFunctionInterpolate()
    {
        // test provisionable variable
        Assert.AreEqual(
            "'Var=${foo}'",
            BicepFunction.Interpolate($"Var={new ProvisioningVariable("foo", typeof(string))}").ToString());

        // test index expression in interpolation
        Assert.AreEqual(
            "'Endpoint=${foo['bar']}'",
            BicepFunction.Interpolate($"Endpoint={
                new IndexExpression(
                    new IdentifierExpression("foo"),
                    "bar")
                }").ToString());
    }
```